Workbooks/Intune Devices/Devices/IntuneDevices.workbook

```diff
@@ -0,0 +1,351 @@
+        "size": 3,
+        "showExportToExcel": true,
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces",
+        "visualization": "piechart",
+        "gridSettings": {
+          "formatters": [
+            {
+              "columnMatch": "OperationCount",
+              "formatter": 3,
+              "formatOptions": {
+                "showIcon": true
+              }
+            }
+          ]
+        }
+      },
+      "customWidth": "50",
+      "name": "Query - DevicesByComplianceState",
+      "styleSettings": {
+        "maxWidth": "50"
+      }
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "IntuneDevices\r\n| where TimeGenerated > ago(1d)\r\n| where Result != \"Completed\"\r\n| join kind = inner (\r\n    IntuneDevices\r\n    | where TimeGenerated > ago(1d)\r\n    | where Result == \"Completed\"\r\n    | summarize arg_max(TimeGenerated, BatchId)\r\n) on BatchId\r\n| summarize arg_max(DeviceId, *) by DeviceId\r\n| summarize count() by ManagedBy",
+        "size": 3,
+        "showExportToExcel": true,
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces",
+        "visualization": "piechart",
+        "gridSettings": {
+          "formatters": [
+            {
+              "columnMatch": "OperationCount",
+              "formatter": 3,
+              "formatOptions": {
+                "showIcon": true
+              }
+            }
+          ]
+        }
+      },
+      "customWidth": "50",
+      "name": "Query - DevicesByManagedBy",
+      "styleSettings": {
+        "maxWidth": "50"
+      }
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "## Andriod Device Count By Version\r\n---"
+      },
+      "customWidth": "50",
+      "name": "Text - AndroidDeviceCountByVersion",
+      "styleSettings": {
+        "maxWidth": "50"
+      }
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "## iOS Device Count By Version\r\n---"
+      },
+      "customWidth": "50",
+      "name": "Text - iOSDeviceCountByVersion",
+      "styleSettings": {
+        "maxWidth": "50"
+      }
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "IntuneDevices\r\n| where TimeGenerated > ago(1d)\r\n| where Result != \"Completed\"\r\n| where OS contains \"Android\"\r\n| join kind = inner (\r\n    IntuneDevices\r\n    | where TimeGenerated > ago(1d)\r\n    | where Result == \"Completed\"\r\n    | summarize arg_max(TimeGenerated, BatchId)\r\n) on BatchId\r\n| summarize arg_max(DeviceId, *) by DeviceId\r\n| summarize DeviceCount = count() by OSVersion\r\n| sort by OSVersion desc\r\n| project AndroidVersion = OSVersion, DeviceCount",
```

Comment:
not an expert on your data or schema, so just a question: can you use `has` in some of the places you are using `contains` ?  `has` uses an index and can be a lot faster than `contains`, if it gets you the right answer

Reply:
Good point. Fixed!